src/storage/src/hummock/sstable_store.rs

```diff
@@ -960,6 +1042,154 @@ impl BlockStream {
     }
 }
 
+/// An iterator that reads the blocks of an SST step by step from a given stream of bytes.
+pub struct BatchBlockStream {
```

Comment:
`BatchBlockStream` and `BlockStream` share a lot of duplicate codes. Given that the implementation is not trivial now, can we keep only one `BlockStream` or at least let them share the same `inner`?

IIUC, the major difference between the two are `BacthBlockStream` will refill block cache while `BlockStream` will not.